src/ansys/aedt/core/modeler/circuits/primitives_maxwell_circuit.py

```diff
@@ -284,3 +284,87 @@ def create_winding(self, name=None, location=None, angle=0, use_instance_id_netl
         )
         id.set_property("Name", name)
         return id
+
+    @pyaedt_function_handler(compname="name")
+    def create_isin(self, name=None, value=1, location=None, angle=0, use_instance_id_netlist=False):
+        """Create a sinusoidal current source.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the current source. The default is ``None``.
```

Comment:
```suggestion
            Name of the current source. The default is ``None`` in which case a random name will be generated.
```

Reply:
Either this or create the component without a name as per default Maxwell circuit editor usage.

Reply:
Hi @gmalinve , I've made all components in 'primitives_maxwell_circuit' take name parameter in same way. So they can accept None (Now without an error) and changed doc to add support for String, Integer and Float. 
This is the same behavior as in the schematic.